envoy/router/router.h

```diff
@@ -996,6 +1027,11 @@ class RouteEntry : public ResponseEntry {
    * @return std::string& the name of the route.
    */
   virtual const std::string& routeName() const PURE;
+
+  /**
+   * @return PathStatsConfig the config needed to generate path level stats.
+   */
+  virtual const absl::optional<PathStatsConfig>& pathStatsConfig() const PURE;
```

Comment:
Should we use Ptr instead of optional reference here?

Reply:
This is fine.